src/sql/workbench/services/objectExplorer/browser/serverTreeDataSource.ts

```diff
@@ -63,6 +63,8 @@ export class ServerTreeDataSource implements IDataSource {
 		} else if (element instanceof TreeNode) {
 			let node = element;
 			if (node.children) {
+				// Grab the latest data from the server of the node's children.
```

Comment:
What happens if an error occurs during refresh? We already have logic to handle an error during the initial resolve - seems like that should probably apply here too.